src/django_components/util/command.py

```diff
@@ -0,0 +1,407 @@
+import sys
+from argparse import Action, ArgumentParser
+from dataclasses import asdict, dataclass
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Literal, Optional, Protocol, Sequence, Type, Union
+
+if TYPE_CHECKING:
+    from argparse import _ArgumentGroup, _FormatterClass
+
+#############################
+# Argparse typing
```

Comment:
This is the file with the main logic around managing and creating commands.

The API for defining our commands is build on top of `argparse`. Django also uses `argparse`, so at least the behaviour is similar.

I wanted the API to be declarative, but the way `ArgumentParser` is defined is that you add arguments by calling [`ArgumentParser.add_argument()`](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_argument).

So I added these dataclasses to be able to define the inputs to functions like `add_argument()` declaratively.